packages/mimir/src/rules/parameter-properties.ts

```diff
@@ -0,0 +1,193 @@
+
+    public apply() {
+        const checkClass = (node: ts.Node): void => {
+            if (
+                (isClassDeclaration(node) || isClassExpression(node)) &&
+                node.members.filter(isConstructorDeclaration).length === 1
+            )
+                this.checkConstructorParameters(node.members.find(isConstructorDeclaration)!, node);
+
+            return ts.forEachChild(node, checkClass);
+        };
+        ts.forEachChild(this.context.sourceFile, checkClass);
+    }
+
+    private get failureString(): string {
+        switch (this.options.mode) {
+            case 'never':
+                return 'Parameter properties have been disallowed.';
+            case 'when-possible':
+                return 'Use parameter properties when possible.';
+            default:
+                return 'Either all constructor parameters must be parameter properties, or none.';
+        }
+    }
+
+    private checkConstructorParameters(
+        construct: ts.ConstructorDeclaration,
+        classNode: ts.ClassDeclaration | ts.ClassExpression,
+    ): void {
+        switch (this.options.mode) {
+            case 'never':
+                for (const param of construct.parameters.filter(isParameterProperty))
+                    this.addFailureAtNode(
+                        param,
+                        this.failureString,
+                        getFixerForDisallowedParameterProp(
+                            construct,
+                            param,
+                            getLineBreakStyle(this.context.sourceFile),
+                        ),
+                    );
+                break;
+            case 'when-possible':
+                for (const param of construct.parameters.filter(
+                    (p: ts.ParameterDeclaration) => !isParameterProperty(p) && canBeParameterProperty(p, construct),
+                ))
+                    this.addFailureAtNode(
+                        param,
+                        this.failureString,
+                        getFixerForLonghandProp(param, construct, classNode),
+                    );
+                break;
+            default:
+                if (construct.parameters.some(isParameterProperty) && !construct.parameters.every(isParameterProperty))
+                    for (const param of construct.parameters.filter(isParameterProperty))
+                        this.addFailure(
+                            construct.parameters.pos,
+                            construct.parameters.end,
+                            this.failureString,
+                            getFixerForDisallowedParameterProp(
+                                construct,
+                                param,
+                                getLineBreakStyle(this.context.sourceFile),
+                            ),
+                        );
+        }
+    }
+}
+
+function canBeParameterProperty(param: ts.ParameterDeclaration, construct: ts.ConstructorDeclaration): boolean {
```

Comment:
this function could simply exclude all parameters where `param.name.kind !== ts.SyntaxKind.Identifier`